packages/playground/app/api/webhook/route.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  setUserNotificationDetails,
+  deleteUserNotificationDetails,
+} from "@/lib/notification";
+import { sendFrameNotification } from "@/lib/notification-client";
+import { http } from "viem";
+import { createPublicClient } from "viem";
+import { optimism } from "viem/chains";
+
+const appName = process.env.NEXT_PUBLIC_ONCHAINKIT_PROJECT_NAME;
+
+const KEY_REGISTRY_ADDRESS = "0x00000000Fc1237824fb747aBDE0FF18990E59b7e";
+
+async function verifyFidOwnership(fid: number, appKey: `0x${string}`) {
+  const client = createPublicClient({
+    chain: optimism,
+    transport: http(),
+  });
+
+  try {
+    const result = await client.readContract({
+      address: KEY_REGISTRY_ADDRESS,
+      abi: [
+        {
+          inputs: [
+            { name: "fid", type: "uint256" },
+            { name: "key", type: "bytes" },
+          ],
+          name: "keyDataOf",
+          outputs: [
+            {
+              components: [
+                { name: "state", type: "uint8" },
+                { name: "keyType", type: "uint32" },
+              ],
+              name: "",
+              type: "tuple",
+            },
+          ],
+          stateMutability: "view",
+          type: "function",
+        },
+      ],
+      functionName: "keyDataOf",
+      args: [BigInt(fid), appKey],
+    });
+
+    return result.state === 1 && result.keyType === 1;
+  } catch (error) {
+    console.error("Key Registry verification failed:", error);
+    return false;
+  }
+}
+
+function decode(encoded: string) {
+  return JSON.parse(Buffer.from(encoded, "base64url").toString("utf-8"));
+}
+
+export async function POST(request: Request) {
+  const requestJson = await request.json();
+
+  const { header: encodedHeader, payload: encodedPayload } = requestJson;
+
+  const headerData = decode(encodedHeader);
+  const event = decode(encodedPayload);
+
+  const { fid, key } = headerData;
+
+  const valid = await verifyFidOwnership(fid, key);
+  console.log("valid", valid);
```

Comment:
nit: I know it's the playground but I feel like this could either be deleted or logged in a more descriptive way (e.g. `Is fid valid: {valid}`)